core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java

```diff
@@ -45,6 +46,7 @@
     private String tokenUrl;
     private String jwksUrl;
     private Optional<String> userinfoUrl = Optional.empty();
+    private String userInfoRequestMethod = POST;
```

Comment:
`userinfoUrl` uses a different casing

Reply:
Since `userinforUrl` is optional, so `userinforRequestMethod` should be optional as well. Configuring:
```properties
http-server.authentication.oauth2.userinfo-request-method=GET
```
shouldn't be allowed without `http-server.authentication.oauth2.userinfo-url` specified.

Reply:
All remarks have been processed. I've changed the casing, made the property optional and added state checks.